CavalcanteLucas/labdale
# Refactor/OL111-message_module

**Link to task:** 
https://labcodes.atlassian.net/browse/OL-111

**How to test:**
- At the welcome page, perform one or more of the following tasks in any sequence successfully, then check if messages reporting success is exhibited as alerts. Those alert messages should be displayed until explicitly dismissed by the user or in case of Login.
  - Register
  - Reset password
- At the dashboard page, perform one or more of the following tasks in any sequence successfully, then check if messages reporting success is exhibited as alerts. Those alert messages should be displayed until explicitly dismissed by the user or in case of Logout.
  - Create TodoList
  - Edit TodoList
  - Delete TodoList
  - Create Todo

**Description of your solution:**
This PR refactor how success messages are handled after authentication related operations (register, and reset password) or CRUD operations over TodoLists and Todos. In this PR, a react component 'Messager' that encapsulates the logic for exhibition of success messages has been implemented. This PR also implements a correspondent redux store 'messagerReducers' to handle state changes due to successful action types. Messages of success are stored in a list of strings in a redux state. Such a list is incremented with a new message whenever a successful action is performed. A message can be removed from this list at any time by the user and only then. This list is emptied out in case of successful login or logout.
